### PR TITLE
Add MLFlow logger

### PR DESCRIPTION
**Pull request by RMeli**
## Description

Add logging with [MLFlow](https://mlflow.org/). Logging includes all input parameters, metrics, and also some artifacts (training and test metrics in CSV file and log file).

### CODE REVIEW

File: gnina/training.py [hunk 1]
```diff
@@ -661,31 +698,28 @@ def log_training_results(trainer):
...
             metrics_train[key].append(value)
 
         # Update LR based on the loss on the training set
         if args.lr_dynamic:
-            loss = mts["Loss (pose)"]
-            try:
-                loss += mts["Loss (affinity)"]
-            except KeyError:
-                # No affinity loss
-                pass
```
Comment (RMeli) [inline gnina/training.py:688]: This `try/except` block was very dangerous. Changing the name of the metric without changing the key here would have failed silently.
